### PR TITLE
Retina for static map services

### LINKED ISSUE #187: Retina with Static Api?
Hello, it seems I can't choose the retina format for my static generated maps.

If I look in the code:

` if auto:
pth = '/{mapid}/geojson({overlay})/auto/{width}x{height}.{fmt}'`

But if I look at the global documentation, the url scheme is `{width}x{height}@2x.{fmt}`

So the is not the possibility to add retina images here?

### LINKED ISSUE #190: StaticStyle.image: twox=True generates a URL that the MapBox API barfs on
```
import mapbox
import os

style = mapbox.StaticStyle(access_token=os.environ['MAPBOX_ACCESS_TOKEN'])
response = style.image(
    username='tomjakubowskiob',
    style_id='cj383j11j00052roea199zqcv',
    zoom=2.0, lon=0.00, lat=0.00,
    pitch=0.0, bearing=0.0,
    width=640,
    height=640,
    twox=True)  # setting this to False also results in success, but w/ no @2x rendering
print response.status_code
print response.request.url
```

On my machine, this prints `404` and then something like `https://api.mapbox.com/styles/v1/tomjakubowskiob/cj383j11j00052roea199zqcv/static/0.0,0.0,2.0,0.0,0.0/640x640%402x?access_token=NOT_TELLING_YOU`

When I run that same URL through curl (being careful to quote the URL at the shell), I also see a 404.

```
$ curl -vo out.png 'https://api.mapbox.com/styles/v1/tomjakubowskiob/cj383j11j00052roea199zqcv/static/0.0,0.0,2.0,0.0,0.0/640x640%402x?access_token=NOT_TELLING_YOU'
...
< HTTP/1.1 404 Not Found
...
```

However, if "unescape" by hand the `%40` to a `@` character in the curl invocation, I finally obtain a successful map image.

```
$ curl -vo out.png 'https://api.mapbox.com/styles/v1/tomjakubowskiob/cj383j11j00052roea199zqcv/static/0.0,0.0,2.0,0.0,0.0/640x640@2x?access_token=NOT_TELLING_YOU'
...
< HTTP/1.1 200 OK
...
```

### PR DESCRIPTION
**Pull request by perrygeo**
Addresses a few issues with our Static and StaticStyle services
* Adds `retina` argument to the Static map service (resolves #187)
* Fixes escaped `@` in URI for StaticStyle (resolves #190)
* Deprecates `StaticStyle`'s current `twox` argument in favor of `retina`. Both will continue to work, use of `twox` will warn, specifying both will raise an exception.

### CODE REVIEW

**Comment by perrygeo** [comment]
@sgillies ready for 👀

File: mapbox/services/static.py [hunk 1]
```diff
@@ -69,19 +69,25 @@ def image(self, mapid, lon=None, lat=None, z=None, features=None,
...
             # No overlay
-            pth = '/{mapid}/{lon},{lat},{z}/{width}x{height}.{fmt}'
+            pth = '/{mapid}/{lon},{lat},{z}/{width}x{height}'
 
         uri = URITemplate(self.baseuri + pth).expand(**values)
+
+        # @2x.format handled separately to avoid HTML escaping the ampersand
+        twox = '@2x' if retina else ''
+        full_fmt = '{0}.{1}'.format(twox, image_format)
+        uri += full_fmt
```
Comment (sgillies) [inline mapbox/services/static.py]: I appreciate the comment there!

File: mapbox/services/static_style.py [hunk 2]
```diff
@@ -91,6 +92,15 @@ def image(self, username, style_id, lon=None, lat=None, zoom=None, features=None
...
 
+        # twox as a deprecated alias for retina
+        if retina is None:
+            if twox is not None:
+                warnings.warn('twox is a deprecated alias for retina', DeprecationWarning)
+                retina = twox
+        else:
+            if twox is not None:
+                raise errors.ValidationError('Conflicting args; Remove twox and use retina')
+
```
Comment (sgillies) [inline mapbox/services/static_style.py]: @perrygeo I've been persuaded by Brett Slatkin (in "Effective Python") that we should draw a line between Python deprecation warnings, which are for core Python developers, and our own deprecation warnings, which are for our SDK users. I'm inclined to switch to a `MapboxDeprecationWarning` that derives from `UserWarning` and isn't ignored by default. What would you think of that?

Comment (perrygeo) [inline mapbox/services/static_style.py]: Great idea. Being able to differentiate core/library/applications has lots of benefits.

**Pull request merged**